es5-shim.js

```diff
@@ -329,6 +360,97 @@ var strSplit = call.bind(StringPrototype.split);
 var strIndexOf = call.bind(StringPrototype.indexOf);
 var push = call.bind(array_push);
 
+// ES-5 15.3.4.4
+// http://es5.github.io/#x15.3.4.4
+var hasToStringTagBasicBug = !(toStr() === '[object Undefined]' && toStr(null) === '[object Null]');
+var hasToStringTagArgumentsBug = toStr(function () {
+  return arguments;
+}) !== '[object Arguments]';
+var hasToStringTagArrayBug = toStr([]) !== '[object Array]';
+var hasToStringTagRegExpBug = toStr(/test/) !== '[object RegExp]';
+var hasToStringTagBug = hasToStringTagBasicBug ||
+    hasToStringTagArgumentsBug ||
+    hasToStringTagArrayBug ||
+    hasToStringTagRegExpBug;
+
+// Other to check:
+// Safari 8 returns 'object' for typed array constructors-
+// PhantomJS 1.9 which returns 'function' for `NodeList` instances.
+
+if (hasToStringTagArgumentsBug) {
+  var isDuckTypeArguments = function (value) {
+    /*jshint eqnull:true */
+    return value != null &&
+        typeof value === 'object' &&
+        owns(value, 'length') &&
+        typeof value.length === 'number' &&
+        value.length >= 0 &&
+        owns(value, 'callee') &&
+        isCallable(value.callee) &&
+        !owns(value, 'arguments');
+
+  };
+}
+
+if (hasToStringTagArrayBug) {
+  var isDuckTypeArray = function (value) {
+    /*jshint eqnull:true */
+    return value != null &&
+        typeof value === 'object' &&
+        owns(value, 'length') &&
+        typeof value.length === 'number' &&
+        value.length >= 0 &&
+        !owns(value, 'callee') &&
+        !owns(value, 'arguments');
+  };
+}
+
+if (hasToStringTagRegExpBug) {
+  var isDuckTypeRegExp = function (value) {
+    /*jshint eqnull:true */
+    if (value != null) {
+      var type = typeof value;
+      if (type === 'object' || type === 'function') {
+        try {
+          regexExec.call(value);
```

Comment:
Not really sure how this can be achieved. If someone deletes `Function.prototype.call` it seems everything is pretty much fubar, whether es5-shim is loaded or not. Did you have something in mind?